Polyfills/Console/Source/Console.cpp

```diff
@@ -36,18 +38,64 @@ namespace
 
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
-        std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
-        {
-            if (i > 0)
-            {
-                ss << " ";
+        
+        std::string formattedString = "";
+        if (info.Length() > 0) {
+            formattedString = info[0].ToString().Utf8Value();
+            // check if this string has substitutions or not
+            std::regex toSub("(%[oOs])|(%(\\d*\\.\\d*)?[dif])");
+            std::smatch matches;
+            
+            // for each argument beyond the first (which is the string itself, try to find a substitution string)
+            for (size_t i = 1; i < info.Length(); i++) {
+                Napi::Value v = info[i];
+
+                // check if there's a corresponding match to this argument
+                if (formattedString.find("%") != std::string::npos && std::regex_search(formattedString, matches, toSub)) {
+                    std::string match = matches[0].str();
```

Comment:
This copies the string for no reason. Do this instead:
```suggestion
                    const std::string& match = matches[0].str();
```